src/pathfind/teleport.hpp

```diff
@@ -136,9 +138,10 @@ class teleport_map {
 	}
 
 private:
-	std::map<map_location, std::set<std::string>> teleport_map_;
-	std::map<std::string, std::set<map_location>> sources_;
-	std::map<std::string, std::set<map_location>> targets_;
+	std::unordered_map<map_location, std::unordered_set<map_location, map_location_hasher>, map_location_hasher> teleport_map_;
+	std::unordered_set<map_location, map_location_hasher> sources_;
+	std::unordered_set<map_location, map_location_hasher> targets_;
```

Comment:
Does anyone know why these were previously a map of id to sets, rather than just a combined set? Does it matter to remove that distinction? I'm really not sure. (I assume this also relates to my `teleport_id` question from earlier.)

Reply:
It should make no difference, the class functions should return the same result.